src/Transit/Internal/FileTransfer.hs

```diff
@@ -66,28 +72,27 @@ send session appid password printHelpFn tfd = do
                        Right (Transit peerAbilities peerHints) -> do
                          -- send offer for the file
                          offerResp <- senderOfferExchange conn filepath
-                         fileBytes <- BS.readFile filepath
                          case offerResp of
                            Left s -> panic s
                            Right _ ->
                              runTransitProtocol peerAbilities peerHints asyncServer
                              (\endpoint -> do
                                  -- 0. derive transit key
                                  let transitKey = MagicWormhole.deriveKey conn (transitPurpose appid)
-                                 -- 1. handshakeExchange
+                                 -- 1. create record keys
+                                     sRecordKey = makeSenderRecordKey transitKey
+                                     rRecordKey = makeReceiverRecordKey transitKey
+                                 -- 2. handshakeExchange
                                  senderHandshakeExchange endpoint transitKey
-                                 -- 2. create record keys
-                                 let sRecordKey = makeSenderRecordKey transitKey
                                  -- 3. send encrypted chunks of N bytes to the peer
-                                 txSha256Hash <- sendRecords endpoint sRecordKey fileBytes
+                                 (txSha256Hash, _) <- C.runConduitRes (sendPipeline filepath endpoint sRecordKey)
                                  -- 4. read a record that should contain the transit Ack.
                                  --    If ack is not ok or the sha256sum is incorrect, flag an error.
-                                 let rRecordKey = makeReceiverRecordKey transitKey
                                  rxAckMsg <- receiveAckMessage endpoint rRecordKey
                                  closeConnection endpoint
                                  case rxAckMsg of
                                    Right rxSha256Hash ->
-                                     when (txSha256Hash /= rxSha256Hash) $
+                                     when (show txSha256Hash /= rxSha256Hash) $
```

Comment:
It seems like using `show` here is a little bit sad.  Dropping down to strings (unicode in this case, I believe?) always feels a bit like a defeat...

I think that `txSha256Hash` is `Hash.Digest SHA256` here (because of the type of `sendPipeline`).  I think that `rxSha256Hash` is `Text` (because of the type of `receiveAckMessage`).  I _guess_ that `rxSha256Hash` is the hex string rendering of the hash and that's what `show txSha256Hash` is going to compute for us?

If so then this seems correct to me, just not ideal.

Reply:
Yes, that was the intention, but I will try to figure out a better way. Thanks.